tensorbay/client/segment.py

```diff
@@ -79,6 +74,14 @@ def __init__(
         self._client = dataset_client._client  # pylint: disable=protected-access
         self._permission: Dict[str, Any] = {"expireAt": 0}
 
+    @staticmethod
+    def _calculate_file_sha1(local_path: str) -> str:
+        sha1_obj = sha1()
+        with open(local_path, "rb") as fp:
+            sha1_obj.update(fp.read())
```

Comment:
better to use buffer to calculate sha1 for less memory usage.
reference: https://stackoverflow.com/questions/22058048/hashing-a-file-in-python